jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Node.java

```diff
@@ -84,6 +84,11 @@ public void insert(ForceObject<T> element) {
     if (log.isTraceEnabled()) {
       log.trace("insert {} into {}", element, this);
     }
+    if (!this.getBounds().contains(element.p.x, element.p.y)) {
+      log.trace("{} outside of spatial bounds {}", element.p, this.getBounds());
+      this.area = area.add(element.p.x, element.p.y);
+    }
+
     if (forceObject == null) {
```

Comment:
if a node is dragged beyond the original bounds, need to update the bounds to include that location